frontend/src/dashboard.js

```diff
@@ -46,623 +44,528 @@ import {
   ResultSummaryWidget
 } from './widgets';
 import { IbutsuContext } from './services/context.js';
+import { useNavigate, useParams } from 'react-router-dom';
 
 
-export class Dashboard extends React.Component {
-  static contextType = IbutsuContext;
-  static propTypes = {
-    eventEmitter: PropTypes.object,
-    navigate: PropTypes.func,
-    params: PropTypes.object,
-  }
+function Dashboard() {
+  const context = useContext(IbutsuContext);
+  const params = useParams();
 
-  constructor(props) {
-    super(props);
-    this.state = {
-      widgets: [],
-      filteredDashboards: [],
-      dashboards: [],
-      selectedDashboard: null,
-      isDashboardSelectorOpen: false,
-      isNewDashboardOpen: false,
-      isWidgetWizardOpen: false,
-      isEditModalOpen: false,
-      editWidgetData: {},
-      dashboardInputValue: '',
-      filterValueDashboard: ''
-    };
-    props.eventEmitter.on('projectChange', (value) => {
-      this.getDashboards(value);
-      this.getDefaultDashboard(value);
-    });
-  }
+  const navigate = useNavigate();
+
+  // dashboard states
+  const [dashboards, setDashboards] = useState([]);
+  const [filteredDBs, setFilteredDBs] = useState([]);
+  const [selectedDB, setSelectedDB] = useState();
+  const [isDBSelectorOpen, setIsDBSelectorOpen] = useState(false);
+  const [isNewDBOpen, setIsNewDBOpen] = useState(false);
+  const [isDeleteDBOpen, setIsDeleteDBOpen] = useState(false);
+
+  // widget states
+  const [widgets, setWidgets] = useState([]);
+  const [isNewWidgetOpen, setIsNewWidgetOpen] = useState(false);
+  const [isEditModalOpen, setIsEditModalOpen] = useState(false);
+  const [editWidgetData, setEditWidgetData] = useState({});
+  const [isDeleteWidgetOpen, setIsDeleteWidgetOpen] = useState(false);
+  const [currentWidget, setCurrentWidget] = useState();
+
+  // typeahead input value states
+  const [selectDBInputValue, setSelectDBInputValue] = useState('');
+  const [filterDBValue, setFilterDBValue] = useState('');
+
+  useEffect(() => {
+    // fetch widgets with the selected dashboard changes
+
+    // TODO is sync necessary with functional component...?
+    //sync_context();
+    console.log('dashboard useEffect selectedDB');
```

Comment:
Guessing you want to leave these then?

Reply:
nah removing, they were for my lifecycle observation while working on it